### DISCUSSION TITLE
Use lib with TDLIB

### DISCUSSION THREAD

**Discussion by snznaota** [eternnoir/pyTelegramBotAPI#1810] [Q&A] [answered]
Please answer these questions before submitting your issue. Thanks!

1. What version of pyTelegramBotAPI are you using?
any
2. What OS are you using?
any
3. What version of python are you using?
any

**tell me, is it possible to work with tdlib?**

**Comment by Badiboy** [comment]
No. Use Telethon.

**Reply by snznaota** [reply]
I already have a project on pytelegrambotapi, how would I quickly implement work with tdlib. The load on the bot is increasing.

**Reply by snznaota** [reply]
блин) русские ж все)

**Reply by snznaota** [reply]
дак вот, есть уже проект на pytelegrambot api - вот думаю как бы быстро и не переписывая всего сделать поддержку tdlib? может быть есть идеи? думал залезть в код pytelegrambotapi и поправить там отправку сообщений через библу с поддержкой tdlib

**Reply by snznaota** [reply]
или может быть есть какие-то рекомендации как мне моего бота под хайлоад наладить?) просто даже в вебхуке max_connections не могу выставить больше 100

**Comment by Badiboy** [comment]
TDLib has nothing with pyTelegramBotAPI. 

You can try ti fully redesign library backend if you want to... May be it will be funny.
max_connections  = 100 - is the Telegtram limitation.

**Reply by snznaota** [reply]
я понимаю, что это ограничение телеграм. использование https://github.com/tdlib/telegram-bot-api снимает это ограничение. 
Я как понимаю, ты писал эту либу? сложно будет там изменить основные методы отправки сообщений?
как я понимаю должно быть все просто, но это не точно.

**Comment by Badiboy** [comment]
>  сложно будет там изменить основные методы отправки сообщений?

Check and try...
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/telebot/apihelper.py

**Reply by snznaota** [reply]
спасибо, покопаюсь)

**Comment by coder2020official** [comment, marked as answer]
Just install telegram bot API server. Launch it, and change telebot.apihelper.API_URL to local URL of local bot API instance running;
Here's a guide to install local bot API server:
https://tdlib.github.io/telegram-bot-api/build.html

**Reply by snznaota** [reply]
спасибо)
